NPE when using advertisedListeners

**Describe the bug**
The broker failed to acquire ownership for namespace bundle when using `advertisedListeners=internal:pulsar://node1:6650,external:pulsar://node1.external:6650` with `external` listener name, error like below:
```
18:56:03.124 [pulsar-ordered-OrderedExecutor-5-0-EventThread] INFO  org.apache.pulsar.broker.cache.LocalZooKeeperCacheService - Successfully copyied bundles data to local zk at /admin/local-policies/public/default
18:56:03.124 [pulsar-ordered-OrderedExecutor-5-0-EventThread] INFO  org.apache.pulsar.broker.cache.LocalZooKeeperCacheService - Successfully created local policies for /admin/local-policies/public/default -- Optional[org.apache.pulsar.common.policies.data.LocalPolicies@a672f53]
18:56:03.181 [pulsar-1-1] INFO  org.apache.pulsar.broker.loadbalance.impl.ModularLoadManagerImpl - 1 brokers being considered for assignment of public/default/0xa0000000_0xb0000000
18:56:03.194 [pulsar-1-1] INFO  org.apache.pulsar.broker.namespace.OwnershipCache - Trying to acquire ownership of public/default/0xa0000000_0xb0000000
18:56:03.324 [pulsar-ordered-OrderedExecutor-5-0-EventThread] INFO  org.apache.pulsar.broker.namespace.OwnershipCache - Successfully acquired ownership of /namespace/public/default/0xa0000000_0xb0000000
18:56:03.325 [pulsar-ordered-OrderedExecutor-5-0-EventThread] WARN  org.apache.pulsar.broker.namespace.NamespaceService - Failed to acquire ownership for namespace bundle public/default/0xa0000000_0xb0000000: java.util.concurrent.CompletionException: java.lang.NullPointerException
18:56:03.326 [pulsar-1-2] INFO  org.apache.pulsar.broker.PulsarService - Loading all topics on bundle: public/default/0xa0000000_0xb0000000
18:56:03.326 [pulsar-ordered-OrderedExecutor-5-0-EventThread] WARN  org.apache.pulsar.broker.lookup.TopicLookupBase - Failed to lookup null for topic persistent://public/default/test with error org.apache.pulsar.broker.PulsarServerException: Failed to acquire ownership for namespace bundle public/default/0xa0000000_0xb0000000
java.util.concurrent.CompletionException: org.apache.pulsar.broker.PulsarServerException: Failed to acquire ownership for namespace bundle public/default/0xa0000000_0xb0000000
        at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:911) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:899) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977) ~[?:1.8.0_112]
        at org.apache.pulsar.broker.namespace.NamespaceService.lambda$searchForCandidateBroker$11(NamespaceService.java:510) ~[org.apache.pulsar-pulsar-broker-2.6.0-ecloud.jar:2.6.0-ecloud]
        at java.util.concurrent.CompletableFuture.uniExceptionally(CompletableFuture.java:870) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(CompletableFuture.java:852) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962) ~[?:1.8.0_112]
        at org.apache.pulsar.broker.namespace.OwnershipCache.lambda$tryAcquiringOwnership$1(OwnershipCache.java:231) ~[org.apache.pulsar-pulsar-broker-2.6.0-ecloud.jar:2.6.0-ecloud]
        at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:656) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962) ~[?:1.8.0_112]
        at org.apache.pulsar.broker.namespace.OwnershipCache$OwnedServiceUnitCacheLoader.lambda$asyncLoad$0(OwnershipCache.java:141) ~[org.apache.pulsar-pulsar-broker-2.6.0-ecloud.jar:2.6.0-ecloud]
        at org.apache.bookkeeper.util.ZkUtils$1.processResult(ZkUtils.java:80) [org.apache.bookkeeper-bookkeeper-server-4.10.0.jar:4.10.0]
        at org.apache.bookkeeper.zookeeper.ZooKeeperClient$10$1.processResult(ZooKeeperClient.java:737) [org.apache.bookkeeper-bookkeeper-server-4.10.0.jar:4.10.0]
        at org.apache.zookeeper.ClientCnxn$EventThread.processEvent(ClientCnxn.java:636) [org.apache.pulsar-pulsar-zookeeper-2.6.0-ecloud.jar:2.6.0-ecloud]
        at org.apache.zookeeper.ClientCnxn$EventThread.run(ClientCnxn.java:510) [org.apache.pulsar-pulsar-zookeeper-2.6.0-ecloud.jar:2.6.0-ecloud]
Caused by: org.apache.pulsar.broker.PulsarServerException: Failed to acquire ownership for namespace bundle public/default/0xa0000000_0xb0000000
        ... 15 more
Caused by: java.util.concurrent.CompletionException: java.lang.NullPointerException
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:659) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632) ~[?:1.8.0_112]
        ... 12 more
Caused by: java.lang.NullPointerException
        at org.apache.pulsar.broker.namespace.NamespaceService.lambda$searchForCandidateBroker$10(NamespaceService.java:500) ~[org.apache.pulsar-pulsar-broker-2.6.0-ecloud.jar:2.6.0-ecloud]
        at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:656) ~[?:1.8.0_112]
        at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:632) ~[?:1.8.0_112]
        ... 12 more
```

--- a/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/validator/MultipleListenerValidator.java
+++ b/pulsar-broker-common/src/main/java/org/apache/pulsar/broker/validator/MultipleListenerValidator.java
@@ -52,7 +52,7 @@ public final class MultipleListenerValidator {
             throw new IllegalArgumentException("`advertisedListeners` and `advertisedAddress` must not appear together");
         }
         if (StringUtils.isBlank(config.getAdvertisedListeners())) {
-            return Collections.EMPTY_MAP;
+            return Collections.emptyMap();
         }
         Optional<String> firstListenerName = Optional.empty();
         Map<String, List<String>> listeners = Maps.newHashMap();
@@ -101,11 +101,7 @@ public final class MultipleListenerValidator {
                     }
                     String hostPort = String.format("%s:%d", uri.getHost(), uri.getPort());
                     reverseMappings.computeIfAbsent(hostPort, k -> Sets.newTreeSet());
-                    Set<String> sets = reverseMappings.get(hostPort);
-                    if (sets == null) {
-                        sets = Sets.newTreeSet();
-                        reverseMappings.put(hostPort, sets);
-                    }
+                    Set<String> sets = reverseMappings.computeIfAbsent(hostPort, k -> Sets.newTreeSet());
                     sets.add(entry.getKey());
                     if (sets.size() > 1) {
                         throw new IllegalArgumentException("must not specify `" + hostPort + "` to different listener.");

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java
@@ -373,8 +373,9 @@ public class NamespaceService {
                             future.completeExceptionally(
                                     new PulsarServerException("the broker do not have " + options.getAdvertisedListenerName() + " listener"));
                         } else {
+                            URI urlTls = listener.getBrokerServiceUrlTls();
                             future.complete(Optional.of(new LookupResult(nsData.get(),
-                                    listener.getBrokerServiceUrl().toString(), listener.getBrokerServiceUrlTls().toString())));
+                                    listener.getBrokerServiceUrl().toString(), urlTls == null ? null : urlTls.toString())));
                         }
                         return;
                     } else {
@@ -471,8 +472,9 @@ public class NamespaceService {
                                         new PulsarServerException("the broker do not have " + options.getAdvertisedListenerName() + " listener"));
                                 return;
                             } else {
+                                URI urlTls = listener.getBrokerServiceUrlTls();
                                 lookupFuture.complete(Optional.of(new LookupResult(ownerInfo, listener.getBrokerServiceUrl().toString(),
-                                        listener.getBrokerServiceUrlTls().toString())));
+                                        urlTls == null ? null : urlTls.toString())));
                                 return;
                             }
                         } else {

